packages/core/src/http-client/csrf-token-header.ts

```diff
@@ -67,14 +74,29 @@ function makeCsrfRequest<T extends HttpRequestConfig>(
       return executeHttpRequest(destination, removeSlash(axiosConfig))
         .then(response => response.headers)
         .catch(e2 => {
-          throw new ErrorWithCause(
-            'Also second try to fetch SRF token failed - No CSRF token fetched.',
-            e2
+          if (hasCsrfToken(e2)) {
+            return e2.response.headers;
+          }
+          logger.warn(
+            new ErrorWithCause(
+              'Also second try to fetch CSRF token failed - No CSRF token fetched.',
+              e2
+            )
           );
+          // todo suggest to disable csrf token handling when the api is implemented
+          return {};
         });
     });
 }
 
+function hasCsrfToken(e1): boolean {
+  return (
+    e1.isAxiosError &&
```

Comment:
why the check on `isAxiosError` is relevant?

Reply:
Because the type of error is any, only if this is an axios error, you can expect the token is located here:
```
error.response.headers['x-csrf-token']
```
This is not good, because axios is bound. 
Alternatives would be, create an abstraction `parseErrorToResponse(error)` in general to allow different errors, and use `parseAxiosErrorToResponse(error)` for this case, which will be similar to the `executeHttpRequest`, where call `executeWithAxios` internally.

Reply:
ah ok.

Reply:
But assuming it is `any` the result of

```ts
    !!e1.response?.headers?.['x-csrf-token']
```

will be the same, right? And if the error is not an axios error but has the csrf token on it, we want this to be true or don't we?

Reply:
I personally would stick to "only consider axios" for the reasons below:
- This workaround is purely developed for the axios issue. There might be some conflicts when used with other client.
- > And if the error is not an axios error but has the csrf token on it, we want this to be true or don't we?
  - If the request uses axios, I would not expect this can happen, because it should be an axios error.
  - If the request does not use axios, I would not know where can I find the token. Therefore, I would like to see some real use case and then improve it.